internal/measurement/measurement.go

```diff
@@ -172,69 +176,55 @@ func Percentage(value, total int64) string {
+// which one can multiple a value in the specified unit by to get the value
+// in terms of the base unit.
+type unit struct {
+	preferredName string
+	names         []string
+	factor        float64
+}
+
+var memoryUnits = []unit{
+	{"B", []string{"b", "byte"}, 1},
+	{"kB", []string{"kb", "kbyte", "kilobyte"}, 1024},
+	{"MB", []string{"mb", "mbyte", "megabyte"}, 1024 * 1024},
+	{"GB", []string{"gb", "gbyte", "gigabyte"}, 1024 * 1024 * 1024},
+	{"TB", []string{"tb", "tbyte", "terabyte"}, 1024 * 1024 * 1024 * 1024},
+	{"PB", []string{"pb", "pbyte", "petabyte"}, 1024 * 1024 * 1024 * 1024 * 1024},
+}
+
 // isMemoryUnit returns whether a name is recognized as a memory size
 // unit.
 func isMemoryUnit(unit string) bool {
-	switch strings.TrimSuffix(strings.ToLower(unit), "s") {
-	case "byte", "b", "kilobyte", "kb", "megabyte", "mb", "gigabyte", "gb":
-		return true
-	}
-	return false
+	_, _, memoryUnit := unitFactor(strings.TrimSuffix(strings.ToLower(unit), "s"), memoryUnits)
+	return memoryUnit
 }
 
 func memoryLabel(value int64, fromUnit, toUnit string) (v float64, u string, ok bool) {
 	fromUnit = strings.TrimSuffix(strings.ToLower(fromUnit), "s")
 	toUnit = strings.TrimSuffix(strings.ToLower(toUnit), "s")
 
-	switch fromUnit {
-	case "byte", "b":
-	case "kb", "kbyte", "kilobyte":
-		value *= 1024
-	case "mb", "mbyte", "megabyte":
-		value *= 1024 * 1024
-	case "gb", "gbyte", "gigabyte":
-		value *= 1024 * 1024 * 1024
-	case "tb", "tbyte", "terabyte":
-		value *= 1024 * 1024 * 1024 * 1024
-	case "pb", "pbyte", "petabyte":
-		value *= 1024 * 1024 * 1024 * 1024 * 1024
-	default:
-		return 0, "", false
+	uf := func(unit string) (string, float64, bool) {
+		return unitFactor(unit, memoryUnits)
 	}
 
-	if toUnit == "minimum" || toUnit == "auto" {
-		switch {
-		case value < 1024:
-			toUnit = "b"
-		case value < 1024*1024:
-			toUnit = "kb"
-		case value < 1024*1024*1024:
-			toUnit = "mb"
-		case value < 1024*1024*1024*1024:
-			toUnit = "gb"
-		case value < 1024*1024*1024*1024*1024:
-			toUnit = "tb"
-		default:
-			toUnit = "pb"
-		}
+	as := func(value float64) (float64, string, bool) {
+		return autoscale(value, memoryUnits)
 	}
 
-	var output float64
-	switch toUnit {
-	default:
-		output, toUnit = float64(value), "B"
-	case "kb", "kbyte", "kilobyte":
-		output, toUnit = float64(value)/1024, "kB"
-	case "mb", "mbyte", "megabyte":
-		output, toUnit = float64(value)/(1024*1024), "MB"
-	case "gb", "gbyte", "gigabyte":
-		output, toUnit = float64(value)/(1024*1024*1024), "GB"
-	case "tb", "tbyte", "terabyte":
-		output, toUnit = float64(value)/(1024*1024*1024*1024), "TB"
-	case "pb", "pbyte", "petabyte":
-		output, toUnit = float64(value)/(1024*1024*1024*1024*1024), "PB"
-	}
-	return output, toUnit, true
+	return convertUnit(value, fromUnit, toUnit, unit{"B", []string{}, 1.0}, uf, as)
+}
+
+var timeUnits = []unit{
+	{"ns", []string{"ns", "nanosecond"}, float64(time.Nanosecond)},
+	{"μs", []string{"μs", "us", "microsecond"}, float64(time.Microsecond)},
```

Comment:
FWIW, I'm a bit scared to have the non-ASCII character in the canonical name.

Reply:
Done (here and elsewhere)